addons/website_slides/views/website_slides_templates_lesson.xml

```diff
@@ -394,43 +410,29 @@
         </div>
     </div>
     <div class="row mt-3 mb-3">
-        <div class="col-12 col-md d-flex align-items-start mb-4 mb-md-0" t-if="len(slide.link_ids)">
-            <span t-if="slide.link_ids" class="text-muted font-weight-bold mr-3">External sources</span>
+        <div class="col-12 col-md d-flex align-items-start mb-4 mb-md-0" t-if="len(slide.slide_resource_ids.filtered(lambda res: res.type == 'url'))">
+            <span t-if="slide.slide_resource_ids.filtered(lambda res: res.type == 'url')" class="text-muted font-weight-bold mr-3">External sources</span>
             <div class="text-muted mr-auto border-left pl-3">
-                <t t-foreach="slide.link_ids" t-as="link">
+                <t t-foreach="slide.slide_resource_ids.filtered(lambda res: res.type == 'url')" t-as="link">
```

Comment:
Here you compute 3 times the same thing, would be nice to add a Qweb variable to simplify the code

Also no need to do `len(recordset)` if you want to check if the recordset is empty`
```suggestion
        <t t-set="resource_urls" t-value="slide.slide_resource_ids.filtered(lambda res: res.type == 'url')"/>
        <div class="col-12 col-md d-flex align-items-start mb-4 mb-md-0" t-if="resource_urls">
            <span t-if="resource_urls" class="text-muted font-weight-bold mr-3">External sources</span>
            <div class="text-muted mr-auto border-left pl-3">
                <t t-foreach="resource_urls" t-as="link">
```